src/Compilers/Core/Portable/SourceGeneration/UserFunction.cs

```diff
@@ -57,58 +57,63 @@ public int GetHashCode([DisallowNull] T obj)
 
     internal static class UserFunctionExtensions
     {
-        internal static Func<TInput, CancellationToken, TOutput> WrapUserFunction<TInput, TOutput>(this Func<TInput, CancellationToken, TOutput> userFunction)
+        internal static Func<TInput, CancellationToken, TOutput> WrapUserFunction<TInput, TOutput>(this Func<TInput, CancellationToken, TOutput> userFunction, bool debugAnalyzers)
         {
             return (input, token) =>
             {
                 try
                 {
                     return userFunction(input, token);
                 }
-                catch (Exception e) when (!ExceptionUtilities.IsCurrentOperationBeingCancelled(e, token))
+                catch (Exception e) when (!debugAnalyzers && !ExceptionUtilities.IsCurrentOperationBeingCancelled(e, token))
```

Comment:
Why do we have a `!debugAnalyzers` here? My mental model is that when the flag is set that essetnially any `when` block where it's checked should be `FailFast` not rethrownig. Is there a subtly I'm missing with this one?

Reply:
We are not rethrowin. If `debugAnalyzers` is `true`, the `catch` won't be considered, so no exceptions are caught here, they instead propagate up and get handled by the `Environment.FailFast`. Could do the fail fast here instead, but not sure what would be the advantage of that?

Reply:
My fear here is that it works the way you describe today. The two `catch` blocks feel a bit distant from each other. Worry that it would be easy for someone to inadvertently insert a `catch` in the middle that didn't respect the flag and broke the desired behavior. Do we have a test for this case hitting the `FailFast` code? If so I would be okay with the current behavior.